### ISSUE TITLE
Consuming 100% of CPU

### ISSUE THREAD

**Issue by solus-hq** [BoostIO/Boostnote#3462]
Hello, noticing this kind of behaviour for the 3rd time for the last week. My notes are just a few (~40 notes)

- OS: macOS 10.15.3 19D76 x86_64
- Host: MacBookPro14,1
- Kernel: 19.3.0
- CPU: Intel i5-7360U (4) @ 2.30GHz
- GPU: Intel Iris Plus Graphics 640
- Memory: 6125MiB / 8192MiB

I'm attaching process sample during CPU clog and some screenshots.

[Sample of Boostnote.txt](https://github.com/BoostIO/boost-releases/files/4179404/Sample.of.Boostnote.txt)

<img width="811" alt="Screenshot 2020-02-09 at 23 12 28" src="https://user-images.githubusercontent.com/26515805/74128382-1c75c000-4bee-11ea-8539-2765297b6232.png">

<img width="811" alt="Screenshot 2020-02-09 at 23 12 32" src="https://user-images.githubusercontent.com/26515805/74128408-2697be80-4bee-11ea-87a6-09646d1b8bfd.png">

**Comment by Flexo013** [comment]
How long are your notes on average and how long in your longest note?

**Comment by solus-hq** [comment]
> How long are your notes on average and how long in your longest note?

Hello! The longest note is 250 lines and contains some advanced markdown formatting, including links, images, etc.

Average notes are somewhat 20-350 lines

**Comment by solus-hq** [comment]
Here's an update as it seems I'm trying to understand high CPU usage.

I'm attaching a note which "possibly" causes Boostnote to hang every time when my mac is OFFLINE, but when I turn wifi back on CPU usage instantly drops from 85-98% to 10-14%. 

I'm attaching two screenshots below, first one is for "offline" operation, second one is "online"

We're offline now:

<img width="962" alt="Screenshot 2020-02-13 at 23 01 49" src="https://user-images.githubusercontent.com/26515805/74473788-444a7980-4eb5-11ea-84a6-b3e5f0c22d6d.png">

And voila, we're back online, CPU is fine:

<img width="962" alt="Screenshot 2020-02-13 at 23 02 23" src="https://user-images.githubusercontent.com/26515805/74473803-4d3b4b00-4eb5-11ea-8bc5-39b84aa21aa7.png">


[article.txt](https://github.com/BoostIO/Boostnote/files/4200751/article.txt)

**Comment by arcturus140** [comment]
can you repeat the test with usage statistics disabled?

**Comment by Flexo013** [comment]
Performing the same online/offline test on Windows with the note you provided yields in less than 1% CPU usage in both cases.

**Comment by solus-hq** [comment]
> can you repeat the test with usage statistics disabled?

Yes, I did it. Disabled usage statistics and it seems (used the app for some 10 minutes) that the app is running smoothly. CPU usage was 20% at maximum. Both offline and online.

**Comment by Flexo013** [comment]
Then usage statistics reporting is definitely the cause of the issue. Introducing some sort of increasing timer or throttle that slows down calls to the remote server should solve this.
